allensdk/brain_observatory/ecephys/ecephys_project_api/ecephys_project_warehouse_api.py

```diff
@@ -248,11 +270,33 @@ def get_unit_analysis_metrics(self, unit_ids=None, ecephys_session_ids=None, ses
             data = json.loads(item.pop("data"))
             item.update(data)
             output.append(item)
+
         
         output = pd.DataFrame(output)
         output.set_index("ecephys_unit_id", inplace=True)
         output.drop(columns="id", inplace=True)
 
+        for colname in output.columns:
+            try:
+                output[colname] = output.apply(lambda row: ast.literal_eval(str(row[colname])), axis=1)
+            except ValueError:
+                pass
+
+        # TODO: remove this
+        # on_screen_rf and p_value_rf were correctly calculated, but switched with one another. This snippet unswitches them.
+        columns = set(output.columns.values.tolist())
+        if "p_value_rf" in columns and "on_screen_rf" in columns:
+
+            pv_is_bool = np.issubdtype(output["p_value_rf"].values[0], bool)
+            on_screen_is_float = np.issubdtype(output["on_screen_rf"].values[0].dtype, np.floating)
+
+            # this is not a good test, but it avoids the case where we fix these in the data for a future release, but 
+            # reintroduce the bug by forgetting to update the code.
+            if pv_is_bool and on_screen_is_float:
+                p_value_rf = output["p_value_rf"].copy()
+                output["p_value_rf"] = output["on_screen_rf"].copy()
+                output["on_screen_rf"] = p_value_rf
```

Comment:
Is there another way to tell that `on_screen_rf` / `p_value_rf` are swapped? Like a version number associated with the data?

Reply:
Good question. There is not a proper date calculated field, however we could add such to the EcephysUnitMetricBundle model in warehouse:
http://stash.corp.alleninstitute.org/projects/TECH/repos/warehouse/browse/warehouse/app/models/ecephys_unit_metric_bundle.rb

Then this check would be "if this bundle does not have a date calculated or it does and the date is prior to when the module fix was deployed".

Reply:
made a jira issue to do the wh work. i think until that is done this is safer, since I don't know exactly when jobs will start to run through updated module.